### PR TITLE
Upgrade to Java 21

### PR DESCRIPTION
**Pull request by vshastri-ch**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,39 +5,40 @@
     <parent>
         <groupId>uk.gov.companieshouse</groupId>
         <artifactId>companies-house-parent</artifactId>
-        <version>2.1.4</version>
+        <version>2.1.5</version>
```
Comment (foman-ch) [inline pom.xml:8]: think there's now a 2.1.6 version

File: pom.xml [hunk 2]
```diff
@@ -5,39 +5,40 @@
...
     <description>Handles accounts XBRL Validator</description>
     <properties>
-        <java.version>11</java.version>
+        <java.version>21</java.version>
 
-        <maven-build-helper-plugin.version>3.2.0</maven-build-helper-plugin.version>
-        <spring-boot-dependencies.version>2.7.10</spring-boot-dependencies.version>
-        <spring-boot-maven-plugin.version>2.7.10</spring-boot-maven-plugin.version>
+        <maven-build-helper-plugin.version>3.6.0</maven-build-helper-plugin.version>
+        <spring-boot-dependencies.version>3.3.0</spring-boot-dependencies.version>
```
Comment (foman-ch) [inline pom.xml:18]: there's now a 3.3.1 version of Spring Boot

**Review by foman-ch — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/companieshouse/account/validator/security/WebSecurityConfig.java [hunk 3]
```diff
@@ -27,25 +29,22 @@ public WebSecurityConfig(UserAuthenticationInterceptor userAuthenticationInterce
...
 
     /**
      * Configure Web Security.
      */
-    @Override
-    public void configure(WebSecurity web) {
-        // Excluding healthcheck endpoint from security filter
-        web.ignoring().antMatchers("/actuator/**");
+    @Bean
+    WebSecurityCustomizer ignoringCustomizer() {
```
Comment (dgurjar-ch) [inline src/main/java/uk/gov/companieshouse/account/validator/security/WebSecurityConfig.java:47]: I think , actual healthCheck end point needs to be configure.

**Review by foman-ch — APPROVED** [review]

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -16,8 +15,14 @@ info.app.name=Account Validator Api
 info.app.description=A backend service allowing users to test XBRL validation on their accounts.
 info.app.version=1.0.0
 info.java-vendor=${java.specification.vendor}
+
 management.endpoints.web.exposure.include=health,mappings
 management.endpoint.shutdown.enabled=true
+management.endpoints.enabled-by-default=false
+management.endpoints.web.base-path=/
+management.endpoints.web.path-mapping.health=/account-validator/validate/healthcheck
```
Comment (dgurjar-ch) [inline src/main/resources/application.properties:23]: I think path don't need two / , Please remove one  backslash(/) .

**Review by dgurjar-ch — APPROVED** [review]

**Pull request merged**